### PR TITLE
Bugfix: fix upload deleted file

### LINKED ISSUE #310: Cannot "Upload to Git" after removing a previously uploaded file
**Describe the bug**

When I click on <kbd>UPLOAD TO GIT</kbd>, all the files in my project are uploaded.

However, if I remove a file after the first upload, and then click on <kbd>UPLOAD TO GIT</kbd> again, the next upload fails.

> An error occured while uploading.

**To Reproduce**

https://github.com/ditrit/leto-modelizer/assets/20184007/fdff99ef-958c-4932-b3eb-81e9bff1d353

There are no errors showing in the console.

**Expected behavior**

I should be able to remove previously uploaded files, and still be able to upload after.

**Application**

- leto version `1.2.0`

### PR DESCRIPTION
**Pull request by aboubakar14**
fix #310

### CODE REVIEW

**Comment by aboubakar14** [comment]
resolve #310

File: tests/unit/components/dialog/GitCommitDialog.spec.js [hunk 1]
```diff
@@ -8,14 +8,17 @@ import FileStatus from 'src/models/git/FileStatus';
...
+const mockNotStagedFileStatus = new FileStatus({ headStatus: 1, workdirStatus: 1, stageStatus: 1 });
+const mockDeletedFileStatus = new FileStatus({ headStatus: 0, workdirStatus: 0, stageStatus: 3 });
 
 jest.mock('src/composables/events/DialogEvent', () => ({
   subscribe: jest.fn(),
 }));
 
 jest.mock('src/composables/Project', () => ({
-  getStatus: jest.fn(() => Promise.resolve([mockStagedFileStatus, mockNotStagedFileStatus])),
+  getStatus: jest.fn(() => Promise.resolve([mockStagedFileStatus,
```
Comment (k-y0u) [inline tests/unit/components/dialog/GitCommitDialog.spec.js:19]: Jump for each element of the array

File: tests/unit/components/dialog/GitCommitDialog.spec.js [hunk 2]
```diff
@@ -50,7 +53,7 @@ describe('Test component: GitCommitDialog', () => {
         expect(wrapper.vm.stagedFiles).toEqual([]);
 
         await wrapper.vm.setFilesStatus({ key: 'GitCommit' });
-        expect(wrapper.vm.stagedFiles).toEqual([mockStagedFileStatus]);
+        expect(wrapper.vm.stagedFiles).toEqual([mockStagedFileStatus, mockDeletedFileStatus]);
```
Comment (k-y0u) [inline tests/unit/components/dialog/GitCommitDialog.spec.js:56]: Jump for each element of the array

File: src/composables/Project.js [hunk 3]
```diff
@@ -437,6 +437,18 @@ export async function appendProjectFile(projectId, file) {
   });
 }
 
+/**
+ * List all the files in the current staging area.
```
Comment (k-y0u) [inline src/composables/Project.js:441]: Get the list of all....

File: src/composables/Project.js [hunk 4]
```diff
@@ -637,11 +637,18 @@ export async function deleteProjectFile(projectId, filePath, deleteParentFolder)
     if (dirFiles.length > 0) {
       await Promise.allSettled(dirFiles.map((fileName) => deleteProjectFile(projectId, `${filePath}/${fileName}`, true)));
     }
+
     await rmDir(`${projectId}/${filePath}`);
   } else {
     await rm(`${projectId}/${filePath}`);
   }
 
+  const listFiles = await gitListFiles(projectId);
```
Comment (k-y0u) [inline src/composables/Project.js:646]: jump line between const and if

File: src/composables/Project.js [hunk 5]
```diff
@@ -725,31 +736,49 @@ export async function gitCommit(projectId, message) {
...
-  await createBranchFrom(
-    project.id,
-    newBranch,
-    currentBranch,
-  );
-
-  await gitCheckout(project.id, newBranch);
-
-  const files = (await getStatus(project.id))
+  const modifiedFiles = files
```
Comment (k-y0u) [inline src/composables/Project.js:742]: delete line between const

File: src/composables/Project.js [hunk 6]
```diff
@@ -725,31 +736,49 @@ export async function gitCommit(projectId, message) {
...
-  await gitCommit(
+  await gitAdd(project.id, modifiedFiles);
+
+  /* Special case for deleted files.
+  Unlike usual git, we CAN NOT add (git add) a deleted file,
+  so here, the idea is to use the remove method of isomorphic-git to stage the
+  deleted files.
+  cf: https://github.com/isomorphic-git/isomorphic-git/issues/1099
+  */
+  const deletedfiles = files.filter((file) => file.isDeleted).map((file) => file.path);
```
Comment (k-y0u) [inline src/composables/Project.js:757]: jumpe after

File: src/composables/Project.js [hunk 7]
```diff
@@ -725,31 +736,49 @@ export async function gitCommit(projectId, message) {
...
+  );
+
+  await createBranchFrom(
+    project.id,
+    newBranch,
+    sha1,
   );
 
+  await gitCheckout(project.id, newBranch);
+
```
Comment (k-y0u) [inline src/composables/Project.js:781]: delete empty line

**Review by k-y0u — CHANGES_REQUESTED** [review]

File: src/composables/Project.js (hunk 7 above)
Comment (aboubakar14) [reply src/composables/Project.js:781]: There is the gitpush after the emtpy line, should I still remove it ?

File: src/composables/Project.js (hunk 3 above)
Comment (aboubakar14) [reply src/composables/Project.js]: Done

File: tests/unit/components/dialog/GitCommitDialog.spec.js (hunk 1 above)
Comment (aboubakar14) [reply tests/unit/components/dialog/GitCommitDialog.spec.js]: Done

Comment (k-y0u) [reply tests/unit/components/dialog/GitCommitDialog.spec.js]: done on top, why not on bottom... don't skip legs day !

File: tests/unit/components/dialog/GitCommitDialog.spec.js (hunk 2 above)
Comment (k-y0u) [reply tests/unit/components/dialog/GitCommitDialog.spec.js]: same !

File: tests/unit/components/dialog/GitStatusDialog.spec.js [hunk 8]
```diff
@@ -60,6 +60,17 @@ describe('Test component: GitStatusDialog', () => {
           new FileStatus({ headStatus: 0, workdirStatus: 2, stageStatus: 2 }),
         ]);
       });
+
+      it('should have two staged file', () => {
```
Comment (justinehell) [inline tests/unit/components/dialog/GitStatusDialog.spec.js:64]: files

File: src/composables/Project.js (hunk 7 above)
Comment (k-y0u) [reply src/composables/Project.js:782]: JUST DO IT !

**Review by justinehell — CHANGES_REQUESTED** [review]

**Review by k-y0u — APPROVED** [review]

**Review by Zorin95670 — APPROVED** [review]

**Review by justinehell — APPROVED** [review]

**Pull request merged**